securedrop/tests/functional/pageslayout/test_source_layout_torbrowser.py

```diff
@@ -16,32 +16,35 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 import pytest
+from tbselenium.utils import disable_js
 
-from tests.functional import journalist_navigation_steps
-from tests.functional import source_navigation_steps
-from tests.functional.functional_test import TORBROWSER
-from . import functional_test
+from tests.functional.app_navigators import SourceAppNagivator
+from tests.functional.pageslayout.functional_test import list_locales
+from tests.functional.pageslayout.screenshot_utils import save_screenshot_and_html
 
 
+@pytest.mark.parametrize("locale", list_locales())
 @pytest.mark.pagelayout
-class TestSourceLayoutTorbrowser(
-        functional_test.FunctionalTest,
-        source_navigation_steps.SourceNavigationStepsMixin,
-        journalist_navigation_steps.JournalistNavigationStepsMixin):
-    default_driver_name = TORBROWSER
-
-    def test_index(self):
-        self.disable_js_torbrowser_driver()
-        self._source_visits_source_homepage()
-        self._screenshot('source-index.png')
-        self._save_html('source-index.html')
-
-    def test_logout(self):
-        self.disable_js_torbrowser_driver()
-        self._source_visits_source_homepage()
-        self._source_chooses_to_submit_documents()
-        self._source_continues_to_submit_page()
-        self._source_submits_a_file()
-        self._source_logs_out()
-        self._screenshot('source-logout_page.png')
-        self._save_html('source-logout_page.html')
+class TestSourceLayoutTorBrowser:
+
+    def test_index_and_logout(self, locale, sd_servers_v2):
+        # Given a source user accessing the app from their browser
+        locale_with_commas = locale.replace("_", "-")
+        with SourceAppNagivator.using_tor_browser_web_driver(
+            source_app_base_url=sd_servers_v2.source_app_base_url,
+            accept_languages=locale_with_commas,
+        ) as navigator:
+
+            # And they have disabled JS in their browser
+            disable_js(navigator.driver)
```

Comment:
I wasn't sure why JS needs to be disabled here.

Reply:
You're right, it doesn't strictly need to be disabled to work, but it does represent the ideal way of how we prefer sources to use SecureDrop at the moment. That would change once we have an instance-independent way to verify that the JS sources would execute is the same that we ship.

(PS: As someone who spent a lot of time on the source interface in recent months, thanks so much for your work!)